tests/Utils/TestRunner.hs

```diff
@@ -114,7 +115,7 @@ stopServer asyncServer = do
 
 run :: Manager -> [Command T.Text] -> IO (HM.HashMap RequestKey (CommandResult Hash))
 run mgr cmds = do
-  sendResp <- doSend mgr $ SubmitBatch cmds
+  sendResp <- doSend mgr . SubmitBatch $ NEL.fromList cmds
```

Comment:
Could also make the input of the function itself be `NonEmpty` .

Reply:
Lets not so we can test the negative case of `[]`

Reply:
`run` seemed pretty central, so I didn't want to make assumptions about its input. Either way, the empty case wasn't being tested, and `NEL.fromList` throws if given empty input :)